core/src/test/java/org/apache/iceberg/TestTransaction.java

```diff
@@ -239,6 +241,43 @@ public void testTransactionConflict() {
         .hasMessage("Injected failure");
   }
 
+  @TestTemplate
+  public void testTransactionFailureBulkDeletionCleanup() {
+    cleanupTables();
```

Comment:
I don't think this is necessary, because this is being called after each test method anyway